mantidproject/mantid
# Indirect - Vesuvio - Add option to calculate Cumulative Angle-Averaged Data within driver script

The VESUVIO instrument scientists have requested for a method by which to increase the information content of NCS data, by providing an option within the VESUVIO driver script to compute Cumulative Angle-Averaged Data (CAAD).

CAAD is calculated by computing the sum of each fit normalised by its integral. Each component of this num (i.e. each fit normalised by its integral), should also be output as a workspace group.

This PR adds the CAAD method to the Vesuvio Driver Script along with a new boolean flag, ```calculate_caad```, which provides users a manner by which to enable/disable the computation of the Cumulative Angle-Averaged Data.

**To test:**
1. Ensure the directory containing the provided IP file is included in the directories MANTID will search.
2. Load the provided script within MANTID's script window.
3. Execute the script.
4. Ensure two group workspaces named ```23271-23263_CAAD_normalised_iteration_1``` and ```23271-23263_CAAD_normalised_iteration_2``` can be found in the ADS.
5. Within these group workspaces, ensure there are three workspaces with the same name of the group preceded by ```_Calc```, ```_GaussianComptonProfile```, ```_GaussianComptionProfile2```.
6. Ensure each of these workspaces has 3 spectra.
7. Ensure two group workspaces named ```23271-23263_CAAD_sum_iteration_1``` and ```23271-23263_CAAD_sum_iteration_2``` can be found in the ADS.
8. Within these group workspaces, ensure there are three workspaces with the same name of the group preceded by ```_Calc```, ```_GaussianComptonProfile```, ```_GaussianComptionProfile2```.
9. Ensure of of these workspaces has a single spectrum.

**Test Data**
[vesuvio_test_files.zip](https://github.com/mantidproject/mantid/files/1745114/vesuvio_test_files.zip)



Fixes #21787. 

**Release Notes** 
Change added to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
